### ISSUE TITLE
Range faceting fails on numeric labels

### ISSUE THREAD

**Issue by pixelmaven** [typesense/typesense#1445]
## Description

When configuring ranges for a numeric facet, using alphabetical characters works great. But I've run into an instance where grouping by "years" makes sense to use numeric characters which is throwing an error.

## Steps to reproduce

This works:

```
facet_by: "year(Older:[0, 1980], Newer:[1980, 2029])",
```

This throws an error:
```
facet_by: "year(Older:[0, 1980], 1980s:[1980, 1990], 1990s:[1990, 2000], 2000s:[2000, 2010], 2010s:[2010, 2020], 2020s:[2020, 2030])"
```

## Expected Behavior

Facets are grouped by numeric labels the same as alphabetical labels

## Actual Behavior

```
Error: 400 - Facet range value is not valid
```

## Metadata

**Typesense Version**: `typesense/typesense:0.26.0.rc41`
`typesense-instantsearch-adapter@2.8.0-3`
`typesense@1.7.2`

**Comment by jasonbosco** [comment]
This is an issue on the Typesense Server side. So I've moved this issue to the main typesense repo, and edited the issue description slightly accordingly.

**Comment by kishorenc** [comment]
Fix available in `0.26.0.rc42`.

**Comment by jasonbosco** [comment]
Released in v26
